gtsfm/common/gtsfm_data.py

```diff
@@ -346,3 +346,43 @@ def filter_landmarks(self, reproj_err_thresh: float = 5) -> "GtsfmData":
                 filtered_data.add_track(track)
 
         return filtered_data
+
+    def align_via_Sim3_to_poses(self, wTi_list_ref: List[Optional[Pose3]]) -> "GtsfmData":
+        """Align estimated, sparse multiview result (self) to a set of reference poses.
+
+        Args:
+            wTi_list_ref: list of reference/target camera poses, ordered by camera index.
+
+        Returns:
+            aligned_data: sparse multiview result that is aligned to the poses above.
+        """
+        # these are the estimated poses (source, to be aligned)
+        wTi_list = self.get_camera_poses()
+
+        # align the poses which are valid (i.e. are not None)
+        # some camera indices may have been lost after pruning to largest connected component, leading to None values
+        wTi_list_aligned, gtSest = geometry_comparisons.align_poses_sim3_ignore_missing(wTi_list_ref, wTi_list)
```

Comment:
`gtSest` is not intuitive. Maybe rename it?

Reply:
Also, I'd prefer if there is no reference to GT in this function.

Reply:
Thanks for the review, @ayushbaid. Yeah the name is a bit weird. Maybe gSe, and I'll add an inline comment explaining what g and e stand for (i.e. GT and estimate)?

Reply:
was the abbreviation the unintuitive part, or something else?

Reply:
If we are using ref, we should use ref instead of GT. I think `sim_transform_to_ref`?

Reply:
@ayushbaid I renamed it to `rSe`, let me know if this is clearer now

Reply:
Hmm not exactly. Can we follow the convention of `<from>_S_<to>`? I feel it `rSe` is too similar to that convention, and it will confuse people. The from-to convention is well established in GTSFM and we can use that.

Reply:
Hmm didn't quite follow. The convention we've been using in GTSAM for Similarity(3) is aSb, where "a" is the "to" frame and "b" is the "from" frame.

Reply:
Yep. Lets use that?

RefSw? From the current world to Ref.